### ISSUE TITLE
CUDA

### ISSUE THREAD

**Issue by zikkuratti** [bupticybee/TexasSolver#181]
If you are considering rewriting the calculation module to use CUDA cores, could you please advise from which file the calculation begins?

**Comment by bupticybee** [comment]
Can it really be done? I remember reading some peper says it's impossible. 

Anyway you should check https://github.com/bupticybee/TexasSolver/blob/master/src/solver/PCfrSolver.cpp ,that's where calculation begin.

**Comment by bupticybee** [comment]
> if you provide me some extra architectural info i can try for a 6 month +- Now i go to research some CUDA algorythm

Yeah I can provide anything you want. Please feel free to contact me anytime.

**Comment by zikkuratti** [comment]
Why on check not null moves he leave solve nodes empty sometimes?

**Comment by bupticybee** [comment]
> Why on check not null moves he leave solve nodes empty sometimes?

What do you mean by this? Can you refer to some specific part of the code?

**Comment by zikkuratti** [comment]
![Untitled](https://github.com/bupticybee/TexasSolver/assets/41235726/0a6e46b4-c70c-4acc-bea8-45cfa49d388d)

**Comment by bupticybee** [comment]
> ![Untitled](https://user-images.githubusercontent.com/41235726/271778426-0a6e46b4-c70c-4acc-bea8-45cfa49d388d.png)

Oh you need to double click this node to expand it. It's a QT bug.

**Comment by DiaMyt** [comment]
Hello @zikkuratti ! 
I noticed that you have started working on implementing the solver using GPU cores. Could you please share information about the current progress? Do you have an idea about the feasibility of successfully implementing this functionality?

**Comment by zikkuratti** [comment]
not yet but soon

**Comment by bupticybee** [comment]
> not yet but soon

Like to know this is still on going! Did you manage to get some speedup?

**Comment by zikkuratti** [comment]
now im fixing some bugs on another project after that i push on CUDA

**Comment by zikkuratti** [comment]
And i think about can we speed up it if after console solving flop to json, we not rebuild turn and river, as new tree but add combinations of turn, cuz flop is 22100 turn ~270 000 river is 2,6 billions of extra

**Comment by bupticybee** [comment]
> I need fast instruction how to make-build this code. No makefile here. I resarch some cuda stuff and need to recompile your code from scratch.

https://github.com/bupticybee/TexasSolver/tree/console

Please check this branch. Console branch has a makefile, and code structure is much simpler.

**Comment by bupticybee** [comment]
> Also i check console solver it boot #####100% 20-30 sec its too loong i wanna try too boost it, need to find it src folder comands list.

There are ways too, could you contact me somehow? Do you have a telegram account? You can find me at https://t.me/bupticybee ,I'll provide whatever help you need.

**Comment by bupticybee** [comment]
Hi a small update, do you want to check on this? https://github.com/bupticybee/TexasSolver/pull/194

You could work with him on this feature? Or did you already do that?
